terraform console crash when showing ephemeral variables

### Terraform Version

```shell
Terraform v1.10.3
on darwin_arm64
```


### Terraform Configuration Files

```
variable "example" {
  type      = string
  default   = "test"
  ephemeral  = true
}
```


### Debug Output

```
% TF_LOG=trace terraform console
2025-01-03T17:13:27.466-0700 [INFO]  Terraform version: 1.10.3
2025-01-03T17:13:27.466-0700 [DEBUG] using github.com/hashicorp/go-tfe v1.70.0
2025-01-03T17:13:27.466-0700 [DEBUG] using github.com/hashicorp/hcl/v2 v2.23.0
2025-01-03T17:13:27.466-0700 [DEBUG] using github.com/hashicorp/terraform-svchost v0.1.1
2025-01-03T17:13:27.466-0700 [DEBUG] using github.com/zclconf/go-cty v1.15.1-0.20241111215639-63279be090d7
2025-01-03T17:13:27.466-0700 [INFO]  Go runtime version: go1.23.3
2025-01-03T17:13:27.466-0700 [INFO]  CLI args: []string{"terraform", "console"}
2025-01-03T17:13:27.466-0700 [TRACE] Stdout is a terminal of width 183
2025-01-03T17:13:27.466-0700 [TRACE] Stderr is a terminal of width 183
2025-01-03T17:13:27.466-0700 [TRACE] Stdin is a terminal
2025-01-03T17:13:27.466-0700 [DEBUG] Attempting to open CLI config file: /Users/dangj/.terraformrc
2025-01-03T17:13:27.466-0700 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2025-01-03T17:13:27.466-0700 [INFO]  Loading CLI configuration from /Users/dangj/.terraform.d/credentials.tfrc.json
2025-01-03T17:13:27.466-0700 [DEBUG] ignoring non-existing provider search directory terraform.d/plugins
2025-01-03T17:13:27.466-0700 [DEBUG] ignoring non-existing provider search directory /Users/dangj/.terraform.d/plugins
2025-01-03T17:13:27.466-0700 [DEBUG] ignoring non-existing provider search directory /Users/dangj/Library/Application Support/io.terraform/plugins
2025-01-03T17:13:27.466-0700 [DEBUG] ignoring non-existing provider search directory /Library/Application Support/io.terraform/plugins
2025-01-03T17:13:27.467-0700 [INFO]  CLI command args: []string{"console"}
2025-01-03T17:13:27.469-0700 [TRACE] Meta.Backend: no config given or present on disk, so returning nil config
2025-01-03T17:13:27.469-0700 [TRACE] Meta.Backend: backend has not previously been initialized in this working directory
2025-01-03T17:13:27.469-0700 [TRACE] Meta.Backend: using default local state only (no backend configuration, and no existing initialized backend)
2025-01-03T17:13:27.469-0700 [TRACE] Meta.Backend: instantiated backend of type <nil>
2025-01-03T17:13:27.470-0700 [DEBUG] checking for provisioner in "."
2025-01-03T17:13:27.470-0700 [DEBUG] checking for provisioner in "/usr/local/bin"
2025-01-03T17:13:27.470-0700 [TRACE] Meta.Backend: backend <nil> does not support operations, so wrapping it in a local backend
2025-01-03T17:13:27.470-0700 [TRACE] backend/local: CLI option -state is overriding state path to terraform.tfstate
2025-01-03T17:13:27.470-0700 [TRACE] backend/local: requesting state manager for workspace "default"
2025-01-03T17:13:27.470-0700 [TRACE] backend/local: state manager for workspace "default" will:
 - read initial snapshot from terraform.tfstate
 - write new snapshots to terraform.tfstate
 - create any backup at terraform.tfstate.backup
2025-01-03T17:13:27.470-0700 [TRACE] backend/local: requesting state lock for workspace "default"
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: preparing to manage state snapshots at terraform.tfstate
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: no previously-stored snapshot exists
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: locking terraform.tfstate using fcntl flock
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: writing lock metadata to .terraform.tfstate.lock.info
2025-01-03T17:13:27.471-0700 [TRACE] backend/local: reading remote state for workspace "default"
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: reading latest snapshot from terraform.tfstate
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: snapshot file has nil snapshot, but that's okay
2025-01-03T17:13:27.471-0700 [TRACE] statemgr.Filesystem: read nil snapshot
2025-01-03T17:13:27.471-0700 [TRACE] backend/local: populating backendrun.LocalRun for current working directory
2025-01-03T17:13:27.472-0700 [TRACE] terraform.NewContext: starting
2025-01-03T17:13:27.472-0700 [TRACE] terraform.NewContext: complete
2025-01-03T17:13:27.472-0700 [DEBUG] Building and walking 'eval' graph
2025-01-03T17:13:27.472-0700 [TRACE] Executing graph transform *terraform.ConfigTransformer
2025-01-03T17:13:27.472-0700 [TRACE] ConfigTransformer: Starting for path:
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.ConfigTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.RootVariableTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.RootVariableTransformer with new graph:
  var.example - *terraform.NodeRootVariable
  ------
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.ModuleVariableTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.ModuleVariableTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.variableValidationTransformer
2025-01-03T17:13:27.473-0700 [TRACE] variableValidationTransformer: adding validation nodes for any existing variable evaluation nodes
2025-01-03T17:13:27.473-0700 [TRACE] variableValidationTransformer: var.example has no validation rules
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.variableValidationTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.LocalTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.LocalTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.OutputTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.OutputTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.AttachResourceConfigTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.AttachResourceConfigTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.AttachStateTransformer
2025-01-03T17:13:27.473-0700 [DEBUG] Not attaching any node states: overall state is nil
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.AttachStateTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.graphTransformerMulti
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.externalProviderTransformer
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.externalProviderTransformer with new graph:
  var.example - *terraform.NodeRootVariable
  ------
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderConfigTransformer
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderConfigTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.MissingProviderTransformer
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.MissingProviderTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.ProviderTransformer
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.ProviderTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Executing graph transform *terraform.PruneProviderTransformer
2025-01-03T17:13:27.473-0700 [TRACE] (graphTransformerMulti) Completed graph transform *terraform.PruneProviderTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.graphTransformerMulti (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.AttachSchemaTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.AttachSchemaTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.ModuleExpansionTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.ModuleExpansionTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.ReferenceTransformer
2025-01-03T17:13:27.473-0700 [DEBUG] ReferenceTransformer: "var.example" references: []
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.ReferenceTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.CloseProviderTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.CloseProviderTransformer (no changes)
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.CloseRootModuleTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.CloseRootModuleTransformer with new graph:
  root - *terraform.nodeCloseModule
    var.example - *terraform.NodeRootVariable
  var.example - *terraform.NodeRootVariable
  ------
2025-01-03T17:13:27.473-0700 [TRACE] Executing graph transform *terraform.TransitiveReductionTransformer
2025-01-03T17:13:27.473-0700 [TRACE] Completed graph transform *terraform.TransitiveReductionTransformer (no changes)
2025-01-03T17:13:27.473-0700 [DEBUG] Starting graph walk: walkEval
2025-01-03T17:13:27.474-0700 [TRACE] vertex "var.example": starting visit (*terraform.NodeRootVariable)
2025-01-03T17:13:27.474-0700 [TRACE] vertex "var.example": belongs to
2025-01-03T17:13:27.474-0700 [TRACE] NodeRootVariable: evaluating var.example
2025-01-03T17:13:27.474-0700 [TRACE] prepareFinalInputVariableValue: preparing var.example
2025-01-03T17:13:27.474-0700 [TRACE] prepareFinalInputVariableValue: var.example has a default value
2025-01-03T17:13:27.474-0700 [TRACE] prepareFinalInputVariableValue: var.example has no defined value
2025-01-03T17:13:27.474-0700 [TRACE] vertex "var.example": visit complete
2025-01-03T17:13:27.474-0700 [TRACE] vertex "root": starting visit (*terraform.nodeCloseModule)
2025-01-03T17:13:27.475-0700 [TRACE] vertex "root": does not belong to any module instance
2025-01-03T17:13:27.475-0700 [TRACE] vertex "root": visit complete
> var.example
2025-01-03T17:13:33.846-0700 [TRACE] statemgr.Filesystem: removing lock metadata file .terraform.tfstate.lock.info
2025-01-03T17:13:33.847-0700 [TRACE] statemgr.Filesystem: unlocking terraform.tfstate using fcntl flock

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
Please report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version, the stack trace
shown below, and any additional information which may help replicate the issue.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

panic: value is marked, so must be unmarked first
goroutine 1 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:26 +0x64
github.com/hashicorp/terraform/internal/logging.PanicHandler()
	github.com/hashicorp/terraform/internal/logging/panic.go:84 +0x184
panic({0x1039e5680?, 0x1040e74b0?})
	runtime/panic.go:785 +0x124
github.com/zclconf/go-cty/cty.Value.assertUnmarked(...)
	github.com/zclconf/go-cty@v1.15.1-0.20241111215639-63279be090d7/cty/marks.go:141
github.com/zclconf/go-cty/cty.Value.AsString({{{0x10412da18?, 0x1400000f8a9?}}, {0x103cdc8e0?, 0x1400092a600?}})
	github.com/zclconf/go-cty@v1.15.1-0.20241111215639-63279be090d7/cty/value_ops.go:1385 +0x54
github.com/hashicorp/terraform/internal/repl.formatMultilineString({{{0x10412da18?, 0x1400000f8a9?}}, {0x103cdc8e0?, 0x1400092a600?}}, 0x0)
	github.com/hashicorp/terraform/internal/repl/format.go:84 +0x38
github.com/hashicorp/terraform/internal/repl.FormatValue({{{0x10412da18, 0x1400000f8a9}}, {0x103cdc8e0, 0x1400092a600}}, 0x0)
	github.com/hashicorp/terraform/internal/repl/format.go:53 +0x498
github.com/hashicorp/terraform/internal/repl.(*Session).handleEval(0x140005238c8, {0x14000523690?, 0x1037a8d98?})
	github.com/hashicorp/terraform/internal/repl/session.go:86 +0x4e0
github.com/hashicorp/terraform/internal/repl.(*Session).Handle(0x140005238c8, {0x14000523690, 0xc})
	github.com/hashicorp/terraform/internal/repl/session.go:43 +0xe8
github.com/hashicorp/terraform/internal/command.(*ConsoleCommand).modeInteractive(0x14000399dc0, 0x140005238c8, {0x104135c30, 0x14000809e00})
	github.com/hashicorp/terraform/internal/command/console_interactive.go:91 +0x3d8
github.com/hashicorp/terraform/internal/command.(*ConsoleCommand).Run(0x14000399dc0, {0x140001bc010?, 0x0?, 0x0?})
	github.com/hashicorp/terraform/internal/command/console.go:184 +0xd08
github.com/hashicorp/cli.(*CLI).Run(0x14000710640)
	github.com/hashicorp/cli@v1.1.6/cli.go:265 +0x488
main.realMain()
	github.com/hashicorp/terraform/main.go:339 +0x18b4
main.main()
	github.com/hashicorp/terraform/main.go:64 +0x1c
```


### Expected Behavior

$ terraform console
> var.example
it should output the variable's value.

### Actual Behavior

It crashes the "terraform console"

### Steps to Reproduce

1. terraform console
2. var.example

### Additional Context

_No response_

### References

_No response_

### Generative AI / LLM assisted development?

_No response_

--- a/internal/repl/format.go
+++ b/internal/repl/format.go
@@ -23,6 +23,9 @@ func FormatValue(v cty.Value, indent int) string {
 	if v.HasMark(marks.Sensitive) {
 		return "(sensitive value)"
 	}
+	if v.HasMark(marks.Ephemeral) {
+		return "(ephemeral value)"
+	}
 	if v.IsNull() {
 		ty := v.Type()
 		switch {

--- a/internal/repl/format_test.go
+++ b/internal/repl/format_test.go
@@ -177,6 +177,10 @@ EOT_`,
 			cty.StringVal("a sensitive value").Mark(marks.Sensitive),
 			"(sensitive value)",
 		},
+		{
+			cty.StringVal("an ephemeral value").Mark(marks.Ephemeral),
+			"(ephemeral value)",
+		},
 	}
 
 	for _, test := range tests {

